compiler/rustc_errors/src/diagnostic_builder.rs

```diff
@@ -13,6 +13,16 @@ use std::marker::PhantomData;
 use std::ops::{Deref, DerefMut};
 use std::thread::panicking;
 
+/// Trait implemented by error types. This should not be implemented manually. Instead, use
+/// `#[derive(Diagnostic)]` -- see [rustc_macros::Diagnostic].
+#[cfg_attr(bootstrap, rustc_diagnostic_item = "SessionDiagnostic")]
+#[cfg_attr(not(bootstrap), rustc_diagnostic_item = "Diagnostic")]
```

Comment:
```suggestion
#[cfg_attr(not(bootstrap), rustc_diagnostic_item = "IntoDiagnostic")]
```

Reply:
Good point, addressed [here](https://github.com/rust-lang/rust/pull/101558/commits/a6ae7f3c554bf2cf804074b4b08c506258853788).

_(Included it in the rebase/fix/etc commit)_